src/foam/dao/MaterializedDAO.js

```diff
@@ -96,8 +96,15 @@ foam.CLASS({
       class: 'Object',
       type: 'foam.nanos.auth.Authorizer',
       name: 'authorizer',
+      documentation: `
+        Checks read access to the sourceDAO.
+        If the class of the materializedDAO is Authorizable, use its custom authorization
+        logic to check access to the sourceDAO.
+        Otherwise, use a StandardAuthorizer to explicitly check the read permission of the
+        sourceDAO objects.
+      `,
       javaFactory: `
-      if ( foam.nanos.auth.Authorizable.class.isAssignableFrom(getSourceDAO().getOf().getObjClass()) ) {
+      if ( foam.nanos.auth.Authorizable.class.isAssignableFrom(getOf().getObjClass()) ) {
```

Comment:
The javaFactory on `permissionPrefix` property should return the sourceDAO.of and the javascript factory is not needed.

Reply:
thanks updated